### PR TITLE
Add new `Style/TernaryOperator` cop

### PR DESCRIPTION
**Pull request by secade**
I use SimpleCov and CodeClimate to help enforce styles on my projects which (as far as I know) have difficulty detecting appropriate coverage for single-line logical branches, like the ternary operator. It would be nice to be able to turn off all uses of ternaries for projects, and may help keeping code golf out of production systems.

As this is likely just a personal preference, I've set this to be disabled by default.

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [ ] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Used the same coding conventions as the rest of the project.
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [ ] All tests(`rake spec`) are passing.
* [ ] The new code doesn't generate RuboCop offenses that are checked by `rake internal_investigation`.
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [x] Updated cop documentation with `rake generate_cops_documentation` (required only when you've added a new cop or changed the configuration/documentation of an existing cop).

[1]: http://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: spec/rubocop/cop/style/ternary_operator_spec.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+          b :
+          c
+    RUBY
+
+    expect(cop.offenses.size).to eq(1)
+  end
+
+  it 'does not accept a single line ternary operator expression' do
+    inspect_source(<<-RUBY.strip_indent)
+      a = cond ? b : c
```
Comment (Drenmi) [inline spec/rubocop/cop/style/ternary_operator_spec.rb]: The parsed AST should be the same regardless of the whitespace, and since the cop uses the AST only, these assertions don't really test any edge cases. 🙂 

Please also use the new `#expect_offense` matcher, e.g.:

```ruby
it 'registers an offense for ternary operators' do
  expect_offense(<<-RUBY.strip_indent)
    a = cond ? b : c
       ^^^^^^^^^^^^ Do not use ternary operators, use if / else instead.
  RUBY
end
```

**Comment by Drenmi** [comment]
@secade: The test failure is some intermittent issue we had with the CI. Your tests look okay. 🙂

File: spec/rubocop/cop/style/ternary_operator_spec.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    RUBY
+
+    expect(cop.offenses.size).to eq(1)
+  end
+
+  it 'registers an offense when everything is on a separate line' do
+    inspect_source(<<-RUBY.strip_indent)
+      a = cond ?
+          b :
+          c
```
Comment (pocke) [inline spec/rubocop/cop/style/ternary_operator_spec.rb]: I think this cop should not add offence to multiline ternary operator. Because coverage is calculated for a ternary operator if the operator is on multiline.

**Comment by secade** [comment]
Thanks for the reviews, I'll make my updates soon. Not sure why I didn't get a notification of the comments, sorry for the delay.

**Comment by bbatsov** [comment]
@secade ping :-)

**Comment by bbatsov** [comment]
Closing due to lack of activity. If anyone wants to pick this up - be my guest.
